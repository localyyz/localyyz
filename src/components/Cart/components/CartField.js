import React from "react";
import {
  View, StyleSheet, Text, TouchableOpacity
} from "react-native";
import {
  Colours, Sizes, Styles
} from "localyyz/constants";
import {
  randInt
} from "localyyz/helpers";

// third party
import * as Animatable from "react-native-animatable";
import MaterialIcon from "react-native-vector-icons/MaterialIcons";

export default class CartField extends React.Component {
  render() {
    return (
      <TouchableOpacity onPress={this.props.onPress} style={[styles.span, this.props.style]}>
        <View style={[Styles.Horizontal, styles.container]}>
          <TouchableOpacity onPress={this.props.onIconPress}>
            <Animatable.View
              animation="zoomIn"
              duration={300}
              delay={randInt(200) + 200}>
              <MaterialIcon name={this.props.icon || "work"} size={Sizes.IconButton / 2} color={this.props.color || Colours.Text} />
            </Animatable.View>
          </TouchableOpacity>
          {this.props.children}
        </View>
      </TouchableOpacity>
    );
  }
}

const styles = StyleSheet.create({
  span: {
    flex: 1
  },

  container: {
    flex: 1,
    paddingLeft: Sizes.InnerFrame,
    paddingVertical: Sizes.InnerFrame / 2
  }
});
