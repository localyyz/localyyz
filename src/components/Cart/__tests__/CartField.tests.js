import React from "react"
import CartField from "../components/CartField"
import renderer from "react-test-renderer"

describe("CartField", () => {
    it("CartField: should render properly", () => {
        let props = {
            icon: "3d-rotation",
            label: "label",
        }
        const rendered = renderer.create(<CartField {...props}/>)
        expect(rendered).toMatchSnapshot()
    })
})