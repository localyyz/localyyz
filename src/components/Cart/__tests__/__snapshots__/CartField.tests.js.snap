// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CartField CartField: should render properly 1`] = `
<View
  accessible={true}
  isTVSelectable={true}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "flex": 1,
      "opacity": 1,
      "padding": 15,
    }
  }
>
  <View
    style={
      Object {
        "alignItems": "center",
        "flexDirection": "row",
        "justifyContent": "flex-start",
        "paddingBottom": 3.75,
      }
    }
  >
    <View
      accessible={true}
      isTVSelectable={true}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "opacity": 1,
        }
      }
    >
      <View
        style={
          Object {
            "opacity": 0,
            "transform": Array [
              Object {
                "scale": 0.3,
              },
            ],
          }
        }
      >
        <Text
          accessible={true}
          allowFontScaling={false}
          ellipsizeMode="tail"
          style={
            Array [
              Object {
                "color": "rgb(30, 30, 30)",
                "fontSize": 10,
              },
              undefined,
              Object {
                "fontFamily": "Material Icons",
                "fontStyle": "normal",
                "fontWeight": "normal",
              },
            ]
          }
        >
          Ó°ç
        </Text>
      </View>
    </View>
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      numberOfLines={1}
      style={
        Array [
          Object {
            "backgroundColor": "rgba(255, 255, 255, 0)",
            "color": "rgb(30, 30, 30)",
            "flex": 1,
            "fontSize": 12,
            "fontWeight": "300",
            "marginLeft": 3.75,
          },
          Object {
            "color": "rgb(30, 30, 30)",
          },
        ]
      }
    >
      LABEL
    </Text>
  </View>
  <View />
</View>
`;
